From 0a21a439c65cbd523c51c220c41f0fbd2d4f972e Mon Sep 17 00:00:00 2001
From: Florian Westphal <fwestpha@redhat.com>
Date: Tue, 2 Sep 2014 10:14:30 -0400
Subject: [net] ipv6: log src and dst along with "udp checksum is 0"

Message-id: <1409652920-431-8-git-send-email-fwestpha@redhat.com>
Patchwork-id: 90173
O-Subject: [PATCH 07/57] [RHEL7.1 net] ipv6: log src and dst along with "udp checksum is 0"
Bugzilla: 1131999
RH-Acked-by: Jiri Pirko <jpirko@redhat.com>
RH-Acked-by: Vlad Yasevich <vyasevic@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>
RH-Acked-by: Jiri Benc <jbenc@redhat.com>

BZ: 1131999

added this to avoid mangling 'net: Implmement RFC 6936 (zero RX csums
or UDP/IPv6)' patch.

commit 84a3e72c3a5c2441be46c4afba95fe5af3c370e7
Author: Bjorn Mork <bjorn@mork.no>
Date:   Tue Feb 25 21:11:02 2014 +0100

    ipv6: log src and dst along with "udp checksum is 0"

    These info messages are rather pointless without any means to identify
    the source of the bogus packets.  Logging the src and dst addresses and
    ports may help a bit.

    Cc: Joe Perches <joe@perches.com>
    Signed-off-by: Bjorn Mork <bjorn@mork.no>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/ipv6/ip6_checksum.c b/net/ipv6/ip6_checksum.c
index 53f28da..c69fe37 100644
--- a/net/ipv6/ip6_checksum.c
+++ b/net/ipv6/ip6_checksum.c
@@ -79,7 +79,9 @@ int udp6_csum_init(struct sk_buff *skb, struct udphdr *uh, int proto)
   /* RFC 2460 section 8.1 says that we SHOULD log
      this error. Well, it is reasonable.
    */
-  LIMIT_NETDEBUG(KERN_INFO "IPv6: udp checksum is 0\n");
+  LIMIT_NETDEBUG(KERN_INFO "IPv6: udp checksum is 0 for [%pI6c]:%u->[%pI6c]:%u\n",
+          &ipv6_hdr(skb)->saddr, ntohs(uh->source),
+          &ipv6_hdr(skb)->daddr, ntohs(uh->dest));
   return 1;
  }
 
-- 
1.7.1